wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolver.swift

```diff
@@ -147,12 +147,37 @@ public final class OneOnOneResolver: OneOnOneResolverInterface {
             throw OneOnOneResolverError.migratorNotFound
         }
 
-        let mlsGroupID = try await migrator.migrateToMLS(
-            userID: userID,
-            in: context
-        )
+        do {
+            let mlsGroupID = try await migrator.migrateToMLS(
+                userID: userID,
+                in: context
+            )
+            await setReadOnly(to: false, forOneOnOneWithUser: userID, in: context)
+            return .migratedToMLSGroup(identifier: mlsGroupID)
+        } catch MigrateMLSOneOnOneConversationError.failedToEstablishGroup(let error) {
+            await setReadOnly(to: true, forOneOnOneWithUser: userID, in: context)
+            throw MigrateMLSOneOnOneConversationError.failedToEstablishGroup(error)
+        } catch {
+            throw error
+        }
+    }
+
+    private func setReadOnly(
+        to readOnly: Bool,
+        forOneOnOneWithUser userID: QualifiedID,
+        in context: NSManagedObjectContext
+    ) async {
+        await context.perform {
+            guard
+                let otherUser = ZMUser.fetch(with: userID, in: context),
+                let conversation = otherUser.oneOnOneConversation,
+                conversation.isForcedReadOnly != readOnly
+            else {
+                return
+            }
 
-        return .migratedToMLSGroup(identifier: mlsGroupID)
+            conversation.isForcedReadOnly = readOnly
```

Comment:
Do we append a system message saying why the conversation is read only?

Reply:
Since we're waiting for feedback from design, I'll resolve this conversation